compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -172,14 +180,69 @@ public void generateMethod(BIRFunction birFunc, ClassWriter cw, BIRPackage birMo
                     asyncDataCollector, types);
         } else {
             genJMethodForBFunc(birFunc, cw, birModule, jvmTypeGen, jvmCastGen, jvmConstantsGen, moduleClassName,
-                    attachedType, asyncDataCollector);
+                    attachedType, asyncDataCollector, false);
+        }
+    }
+
+    public void genJMethodWithBObjectMethodCall(BIRFunction func, ClassWriter cw, BIRPackage module,
+                                                JvmTypeGen jvmTypeGen, JvmCastGen jvmCastGen,
+                                                JvmConstantsGen jvmConstantsGen, String moduleClassName,
+                                                AsyncDataCollector asyncDataCollector,
+                                                String splitClassName) {
+        BIRVarToJVMIndexMap indexMap = new BIRVarToJVMIndexMap();
+        indexMap.addIfNotExists(OBJECT_SELF_INSTANCE, symbolTable.anyType);
+        indexMap.addIfNotExists(STRAND, symbolTable.stringType);
+        String funcName = func.name.value;
+        BType retType = getReturnType(func);
+        String desc = JvmCodeGenUtil.getMethodDesc(func.type.paramTypes, retType);
+        MethodVisitor mv = cw.visitMethod(Opcodes.ACC_PUBLIC, funcName, desc, null, null);
+        mv.visitCode();
+        Label methodStartLabel = new Label();
+        mv.visitLabel(methodStartLabel);
+        JvmInstructionGen instGen = new JvmInstructionGen(mv, indexMap, module.packageID, jvmPackageGen, jvmTypeGen,
+                jvmCastGen, jvmConstantsGen, asyncDataCollector, types);
+        mv.visitVarInsn(ALOAD, 1); // load strand
+        mv.visitVarInsn(ALOAD, 0); // load self
+        String encodedMethodName = Utils.encodeFunctionIdentifier(funcName);
+        for (BIRNode.BIRFunctionParameter parameter : func.parameters) {
+            instGen.generateVarLoad(mv, parameter, indexMap.addIfNotExists(parameter.name.value, parameter.type));
+        }
+        String methodDesc = JvmCodeGenUtil.getMethodDesc(func.type.paramTypes, retType, moduleClassName);
+        mv.visitMethodInsn(INVOKESTATIC, splitClassName, encodedMethodName, methodDesc, false);
+        Label methodEndLabel = new Label();
+        mv.visitLabel(methodEndLabel);
+        generateReturnTermFromType(retType, mv);
+        mv.visitLocalVariable(OBJECT_SELF_INSTANCE, GET_BOBJECT, null, methodStartLabel, methodEndLabel, 0);
+        mv.visitLocalVariable(STRAND_LOCAL_VARIABLE_NAME, GET_STRAND, null, methodStartLabel, methodEndLabel, 1);
+        for (BIRNode.BIRFunctionParameter parameter : func.parameters) {
+            String metaVarName = parameter.metaVarName;
+            if (isValidArg(parameter) && isCompilerAddedVars(metaVarName)) {
```

Comment:
Why we have this condition?

Reply:
In `MethodGen.createLocalVariableTable` method we only do `mv.visitLocalVariable` for this kind of variables ([check1](https://github.com/ballerina-platform/ballerina-lang/blob/25b62e520d56e41d8ccbfccad539e54bf0b5de7e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java#L1099), [check2](https://github.com/ballerina-platform/ballerina-lang/blob/25b62e520d56e41d8ccbfccad539e54bf0b5de7e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java#L1086)). Hence I added the same check here also.